src/ensembl/src/content/app/species-selector/state/speciesSelectorActions.ts

```diff
@@ -158,42 +206,129 @@ export const handleSelectedSpecies: ActionCreator<
   dispatch(fetchAssemblies(genome_id));
 };
 
-export const commitSelectedSpecies = createStandardAction(
-  'species_selector/commit_selected_species'
-)();
+export const commitSelectedSpecies = createAction(
+  'species_selector/commit_selected_species',
+  (resolve) => {
+    return (itemName: string, committedSpecies: CommittedItem[]) => {
+      return resolve(
+        committedSpecies,
+        buildAnalyticsObject({
+          category: categories.ADD_SPECIES,
+          label: itemName,
+          action: 'select'
+        })
+      );
+    };
+  }
+);
 
 export const commitSelectedSpeciesAndSave: ActionCreator<
   ThunkAction<void, any, null, Action<string>>
 > = () => (dispatch, getState) => {
-  dispatch(commitSelectedSpecies());
   const committedSpecies = getCommittedSpecies(getState());
+  const selectedItem = getSelectedItem(getState());
+
+  if (!selectedItem) {
+    return;
+  }
+
+  const newCommittedSpecies = [
+    ...committedSpecies,
+    buildCommittedItem(selectedItem)
+  ];
+
+  const selectedItemName =
+    selectedItem.common_name || selectedItem.scientific_name;
+
+  dispatch(commitSelectedSpecies(selectedItemName, newCommittedSpecies));
+
   speciesSelectorStorageService.saveSelectedSpecies(committedSpecies);
 };
 
-export const toggleSpeciesUse = createStandardAction(
-  'species_selector/toggle_species_use'
-)<string>();
+export const toggleSpeciesUse = createAction(
+  'species_selector/toggle_species_use',
+  (resolve) => {
+    return (
+      speciesName: string,
+      currentStatus: string,
```

Comment:
Can't this be retrieved from the committedSpecies object (committedSpecies.isEnabled ? 'use' : 'do_not_use' in the body of the function)? Why the extra parameter?

Reply:
Same goes for the speciesName argument

Reply:
It can be but we have to loop through the committedSpecies again to find out which species was toggled.

Reply:
I see. I missed the fact that committedSpecies is an array of species, not a single species 😄